### ISSUE TITLE
UI User Issue

### ISSUE THREAD

**Issue by donavanbecker** [SeydX/homebridge-camera-ui#35]
![image](https://user-images.githubusercontent.com/9875439/96213544-e340dd00-0f3e-11eb-938d-287960e749a8.jpeg)

**Comment by donavanbecker** [comment]
Tried setup system type in new username and password and get this after password change

**Comment by SeydX** [comment]
Hey,

thats strange, i tried again with a fresh installation and changed username and password to something else and it worked

password should be ‚admin‘ and you should give a new password !== admin

**Comment by SeydX** [comment]
![F3EC9605-BEC8-400F-AFA4-F9BDE4F62613](https://user-images.githubusercontent.com/34152761/96214444-c37bda00-0f7b-11eb-8656-dac7b1d501b8.jpeg)

password = old password = admin

new password != admin

**Comment by nexusle** [comment]
Same here...

changed admin to a new user name and the password (admin) to a new one. After this, the new user isn't authorized

**Comment by nexusle** [comment]
With Debug-Mode on:

```
POST / 200 245.800 ms - 91
GET /dashboard 401 410.232 ms - -
GET /stylesheets/modules/bootstrap.min.css 304 6.300 ms - -
GET /stylesheets/style.css 304 4.652 ms - -
GET /fonts/fontawesome.css 304 2.742 ms - -
GET /javascripts/modules/jquery.min.js 304 5.728 ms - -
GET /javascripts/modules/i18next.min.js 304 4.138 ms - -
GET /images/web/logo_transparent-256.png 304 9.689 ms - -
GET /javascripts/modules/velocity.min.js 304 7.092 ms - -
GET /javascripts/modules/velocity.ui.min.js 304 5.538 ms - -
GET /javascripts/modules/jquery.visible.min.js 304 14.089 ms - -
GET /javascripts/modules/socket.io.slim.js 304 12.437 ms - -
GET /javascripts/modules/safearea.min.js 304 11.010 ms - -
GET /javascripts/utils/detectswipe.js 304 9.526 ms - -
GET /javascripts/modules/bootstrap.bundle.min.js 304 8.082 ms - -
GET /plugins/toast/toast.min.js 304 5.533 ms - -
GET /javascripts/main.js 304 4.012 ms - -
```

**Comment by monza258** [comment]
> With Debug-Mode on:
> 
> 
> 
> ```
> 
> POST / 200 245.800 ms - 91
> 
> GET /dashboard 401 410.232 ms - -
> 
> GET /stylesheets/modules/bootstrap.min.css 304 6.300 ms - -
> 
> GET /stylesheets/style.css 304 4.652 ms - -
> 
> GET /fonts/fontawesome.css 304 2.742 ms - -
> 
> GET /javascripts/modules/jquery.min.js 304 5.728 ms - -
> 
> GET /javascripts/modules/i18next.min.js 304 4.138 ms - -
> 
> GET /images/web/logo_transparent-256.png 304 9.689 ms - -
> 
> GET /javascripts/modules/velocity.min.js 304 7.092 ms - -
> 
> GET /javascripts/modules/velocity.ui.min.js 304 5.538 ms - -
> 
> GET /javascripts/modules/jquery.visible.min.js 304 14.089 ms - -
> 
> GET /javascripts/modules/socket.io.slim.js 304 12.437 ms - -
> 
> GET /javascripts/modules/safearea.min.js 304 11.010 ms - -
> 
> GET /javascripts/utils/detectswipe.js 304 9.526 ms - -
> 
> GET /javascripts/modules/bootstrap.bundle.min.js 304 8.082 ms - -
> 
> GET /plugins/toast/toast.min.js 304 5.533 ms - -
> 
> GET /javascripts/main.js 304 4.012 ms - -
> 
> ```

There is no bug in the log. 

You can set in config-ui-x the reset master credentials. 
Reboot and unset

And check again to login. 

username:admin
new Username like: Yourname
Old Password: admin
New Password: NEWPASSWORD

**Comment by nexusle** [comment]
I think `GET /dashboard 401 410.232 ms - -` indicates a authorization issue

I've done the reset multiple times. Everytime with the same result.

**Comment by SeydX** [comment]
I will check this again

i can not reproduce it :/

**Comment by SeydX** [comment]
![089BD30C-29DB-409F-8A36-7601C661DC4F](https://user-images.githubusercontent.com/34152761/96258768-ccd96680-0fbc-11eb-8024-cd0aa520955b.png)

@nexusle @donavanbecker 

Can u set DEBUG=CameraUI* in your environment file pls?

U can find it under homebridge settings (config ui x) or normally (etc/default/homebridge)

**Comment by donavanbecker** [comment]
@SeydX, here is the log and also my config:
```
[10/16/2020, 07:48:59] [CameraUI] Starting User Interface
[10/16/2020, 07:48:59] [CameraUI] CameraUI is listening on port 8585
2020-10-16T12:48:58.704Z CameraUI MQTT for motion handling not active.
2020-10-16T12:48:58.705Z CameraUI HTTP Server for motion handling not active.
2020-10-16T12:48:58.705Z CameraUI Missing interface name in config.json - Looking for interface names..
2020-10-16T12:48:58.863Z CameraUIInterface Configuring database
2020-10-16T12:48:59.091Z CameraUIInterface Configuring cameras
2020-10-16T12:48:59.094Z CameraUIInterface Configuring app
2020-10-16T12:48:59.148Z CameraUIInterface Configuring server
2020-10-16T12:48:59.150Z CameraUIInterface Configuring socket
2020-10-16T12:48:59.164Z CameraUIInterface Configuring clear timer
2020-10-16T12:48:59.290Z CameraUIInterface Configuring handler
2020-10-16T12:48:59.291Z CameraUIInterface Configuring webhook
2020-10-16T12:52:13.099Z CameraUIInterface password (Master): Successfully logged in!
2020-10-16T12:52:13.101Z CameraUIInterface password (Master): Your session expires in 1 hour(s)
2020-10-16T12:52:13.579Z CameraUIApp UnauthorizedError: Unauthorized
```
config:
```
{
            "name": "CameraUI",
            "port": 8585,
            "auth": "form",
            "language": "auto",
            "debug": false,
            "reset": false,
            "mqtt": {
                "active": false
            },
            "http": {
                "active": false,
                "localhttp": false
            },
            "ssl": {
                "active": false
            },
            "options": {
                "videoProcessor": "ffmpeg"
            },
            "cameras": [
                {
                    "name": "Kitchen UI",
                    "manufacturer": "Wyze",
                    "model": "Cam v2",
                    "serialNumber": "xxxx",
                    "firmwareRevision": "xxxx",
                    "motion": true,
                    "videoConfig": {
                        "source": "-i rtsp://xxxx: xxxx@192.168.7.88/live",
                        "stillImageSource": "-i rtsp://xxxx: xxxx@192.168.7.88/live",
                        "vcodec": "copy",
                        "audio": true,
                        "debug": false
                    }
                },
                {
                    "name": "Living Room UI",
                    "manufacturer": "Wyze",
                    "model": "Cam Pan",
                    "serialNumber": "xxxx",
                    "firmwareRevision": "xxxx",
                    "motion": true,
                    "videoConfig": {
                        "source": "-i rtsp://xxxx: xxxx@192.168.7.85/live",
                        "stillImageSource": "-i rtsp://xxxx: xxxx@192.168.7.85/live",
                        "vcodec": "copy",
                        "audio": true,
                        "debug": false
                    }
                }
            ],
            "platform": "CameraUI"
        }
```

**Comment by nexusle** [comment]
Done. You see, that login is successful. It seem's that the user have no rights to see the UI...

`2020-10-16T13:13:08.300Z CameraUIInterface XXXXX (Master): Successfully logged in!
2020-10-16T13:13:08.303Z CameraUIInterface XXXXX (Master): Your session expires in 1 hour(s)
POST / 200 275.289 ms - 92
2020-10-16T13:13:08.393Z CameraUIApp UnauthorizedError: Unauthorized
GET /dashboard 401 406.092 ms - -`

**Comment by SeydX** [comment]
@donavanbecker @nexusle 

Can u show me your database file pls?

Located under config path/db/configui.db.json

(eg var/lib/homebridge/db/configui.db.json)

**Comment by donavanbecker** [comment]
<img width="424" alt="Screen Shot 2020-10-16 at 8 42 14 AM" src="https://user-images.githubusercontent.com/9875439/96265772-7ef83a80-0f8b-11eb-96df-b3cef93a9651.png">

**Comment by SeydX** [comment]
nah users folder is only for profile pics

The users are in configui.db.json at the bottom under „users“

can u post mehr your configui.db.json pls

**Comment by donavanbecker** [comment]
```
{
  "settings": {
    "profile": {
      "logoutTimer": "1"
    },
    "general": {
      "darkmode": false,
      "atHome": false,
      "exclude": [],
      "rooms": [
        "Standard"
      ]
    },
    "dashboard": {
      "refreshTimer": "60",
      "cameras": {}
    },
    "cameras": {},
    "recordings": {
      "active": true,
      "type": "Snapshot",
      "timer": "10",
      "path": "/Users/donavanbecker/.homebridge/recordings",
      "removeAfter": "7"
    },
    "notifications": {
      "clearTimer": "3",
      "clearBanner": "5"
    },
    "telegram": {
      "active": false,
      "token": "",
      "chatID": "",
      "motionOn": "",
      "cameras": {}
    },
    "webhook": {
      "active": false,
      "cameras": {}
    },
    "camviews": {
      "refreshTimer": "60",
      "cameras": {}
    },
    "webpush": {
      "pub_key": "BNO_Bxr4cTI6ntjRJTVyITl1Ac0oR0RkdblkjGnyUBwlzUh8Jhgqi2f1NZK6meZGA2-CcNFE4N3jpv2qpCs6qy8",
      "priv_key": "eljdjEcgYmwGp0LlBiXU6K8_OibWBndvRVGui8DoN6U",
      "subscription": false
    }
  },
  "notifications": {
    "lastFive": {},
    "lastNotifications": {},
    "nots": []
  },
  "users": [
    {
      "id": "80e69ecc-0ab2-4735-b689-8320a8dd1c8c",
      "username": "donavanbecker",
      "role": "Master",
      "photo": "/images/user/anonym.png",
      "changed": "yes",
      "hashedpassword": "$2b$10$jvbGepoTkSJQ17kO72WjU.xFogXqMrFTb.IK/8j3RMcVMQ/kWlKh."
    }
  ]
}
```

**Comment by SeydX** [comment]
Strange, it seems you changed ur credentials successfully...

Hmm

**Comment by SeydX** [comment]
I will publish a new beta version with more debug and some fixes, wait a moment pls

**Comment by donavanbecker** [comment]
@SeydX nice thing about config ui x, is that I can install that beta right from the config!

**Comment by SeydX** [comment]
@donavanbecker @nexusle 

v3.0.2-beta.1 is up could u try pls?

after installation, pls remove ur **configdb.ui.json** file located under **configpath/db/configui.db.json** and restart homebridge

dont forget to **activate** debug under settings and also add **DEBUG=ConfigUI*** in your environment file

**Comment by SeydX** [comment]
After restart, login again with 

username: admin
password: admin

Need the entire log

Thx

**Comment by donavanbecker** [comment]
```
2020-10-16T14:35:44.107Z CameraUIInterface Configuring cameras
2020-10-16T14:35:44.110Z CameraUIInterface Configuring app
2020-10-16T14:35:44.156Z CameraUIInterface Configuring server
2020-10-16T14:35:44.157Z CameraUIInterface Configuring socket
2020-10-16T14:35:44.169Z CameraUIInterface Configuring clear timer
2020-10-16T14:35:44.253Z CameraUIInterface Configuring handler
2020-10-16T14:35:44.254Z CameraUIInterface Configuring webhook
[10/16/2020, 09:35:44] [CameraUI] Starting User Interface
[10/16/2020, 09:35:44] [CameraUI] Master credentials resetted. Please turn off "Reset"!
[10/16/2020, 09:35:44] [CameraUI] CameraUI is listening on port 8585
[10/16/2020, 09:35:59] [Config] Changes to config.json saved.
2020-10-16T14:36:01.730Z CameraUIInterface {
  userID: false,
  message: 'ensure admin',
  url: '/',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:01.732Z CameraUIInterface {
  userID: false,
  message: 'session redirect',
  url: '/',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:16.309Z CameraUIInterface {
  userID: false,
  message: 'ensure admin',
  url: '/',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:16.310Z CameraUIInterface {
  userID: false,
  message: 'session redirect',
  url: '/',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:16.314Z CameraUIInterface Credentials not changed! Please change your credentials!
2020-10-16T14:36:16.697Z CameraUIInterface {
  userID: false,
  message: 'ensure admin',
  url: '/change',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:16.698Z CameraUIInterface {
  userID: false,
  message: 'session redirect',
  url: '/change',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:31.915Z CameraUIInterface {
  userID: false,
  message: 'ensure admin',
  url: '/change',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:31.917Z CameraUIInterface {
  userID: false,
  message: 'session redirect',
  url: '/change',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:31.921Z CameraUIInterface Changing username for Master
2020-10-16T14:36:31.923Z CameraUIInterface Changing password for Master
2020-10-16T14:36:31.926Z CameraUIInterface Changing changed for Master
2020-10-16T14:36:32.134Z CameraUIInterface Changing hashedpassword for donavanbecker
2020-10-16T14:36:32.136Z CameraUIInterface Removing password for donavanbecker
2020-10-16T14:36:32.139Z CameraUIInterface donavanbecker (Master): Successfully logged in!
2020-10-16T14:36:32.141Z CameraUIInterface donavanbecker (Master): Your session will not expire!
2020-10-16T14:36:32.525Z CameraUIInterface {
  userID: false,
  message: 'ensure admin',
  url: '/dashboard',
  authenticated: false,
  noAuth: false
}
2020-10-16T14:36:32.531Z CameraUIApp UnauthorizedError: Unauthorized
```

**Comment by SeydX** [comment]
i can reproduce it now thanks, it seems it occures only if "reset" is true in config?

**Comment by nexusle** [comment]
No. After I'd updated from v2.x to 3.0.1 I must change the user. After this the issue came up. I resetted the credantials (with the config switch) later and set it back after the reboot. 

Sorry, I'm at this time not at home and can't provide log files.

**Comment by SeydX** [comment]
@nexusle @donavanbecker 

v3.0.2-beta.2 published. Can u try it pls?

``sudo npm i -g --unsafe-perm homebridge-camera-ui@beta``

**Comment by nexusle** [comment]
Works now *thumbs up* :-)

**Comment by SeydX** [comment]
@nexusle @donavanbecker 

Dont forget to set a socketPort for the cameras to activate the livestream

See [Example](https://github.com/SeydX/homebridge-camera-ui/blob/master/README.md#livestream)

**Comment by SeydX** [comment]
Should be fixed with v3.0.2

Feel free to reopen if you have still probs
